superset/models/core.py

```diff
@@ -761,24 +761,26 @@ def wrapper(*args, **kwargs):
             post_data = request.form or {}
             d.update(post_data)
             d.update(kwargs)
-            slice_id = d.get('slice_id', 0)
+            slice_id = d.get('slice_id') 
+            form_data = d.get('form_data')
+
             try:
-                slice_id = int(slice_id) if slice_id else 0
-            except ValueError:
+                slice_id = int(d.get('slice_id') or json.loads(d.get('form_data')).get('slice_id'))
```

Comment:
Isn't this ugly? You have slice_id and form_data already unpacked before and you are not using them here. That would make the code a lot more readable.

Also blind exceptions are an antipattern. If you want to go this route just catch TypeError and ValueError